Classes/common/touchdb/TD_Database+Attachments.h

```diff
@@ -55,16 +58,16 @@ typedef enum { kTDAttachmentEncodingNone, kTDAttachmentEncodingGZIP } TDAttachme
                             encoding:(TDAttachmentEncoding *)outEncoding
                               status:(TDStatus *)outStatus;
 
-/** Returns the location of an attachment's file in the blob store. */
-- (NSString *)getAttachmentPathForSequence:(SequenceNumber)sequence
-                                     named:(NSString *)filename
-                                      type:(NSString **)outType
-                                  encoding:(TDAttachmentEncoding *)outEncoding
-                                    status:(TDStatus *)outStatus;
+/** Returns the blob for an attachment in the blob store. */
+- (id<CDTBlobReader>)getAttachmentBlobForSequence:(SequenceNumber)sequence
+                                            named:(NSString *)filename
+                                             type:(NSString **)outType
+                                         encoding:(TDAttachmentEncoding *)outEncoding
+                                           status:(TDStatus *)outStatus;
 
 /** Uses the "digest" field of the attachment dict to look up the attachment in the store and return
- * a file URL to it. DO NOT MODIFY THIS FILE! */
-- (NSURL *)fileForAttachmentDict:(NSDictionary *)attachmentDict;
+ * its blob. */
+- (id<CDTBlobReader>)blobForAttachmentDict:(NSDictionary *)attachmentDict;
```

Comment:
I wonder if there should be separate reader/writer classes to prevent modification of the underlying blob data by people asking for `reader` objects.